src/components/tables/RowActions/ProgressDialog.tsx

```diff
@@ -24,22 +24,29 @@ function ProgressDialog({
     finished,
     renderComponent,
 }: Props) {
+    const inProgressCount = useRef(selectedEntities.length);
     const [done, setDone] = useState(false);
-    const [finishedEntities, setFinishedEntities] = useState<any[]>([]);
-
-    const onFinish = (response: any) => {
-        finishedEntities.push(response);
-        setFinishedEntities(finishedEntities);
-        if (finishedEntities.length === selectedEntities.length) {
-            setDone(true);
-        }
-    };
+    const [finishedCount, setFinishedCount] = useState(0);
 
     const onClose = () => {
-        setFinishedEntities([]);
+        setFinishedCount(0);
         finished();
     };
 
+    const onFinish = useCallback(() => {
+        if (done) {
+            return;
+        }
+
+        setFinishedCount((prev) => prev + 1);
+    }, [done]);
+
+    useEffect(() => {
+        if (finishedCount === inProgressCount.current) {
+            setDone(true);
+        }
+    }, [finishedCount]);
```

Comment:
This really feels like it should be in an effect as the other way of doing it in the click was kind of janky.